### PR TITLE
[conditional bps] cleanup cbs

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #1097

### Summary of Changes

lots of cleanups
* clicking on the gutter when the panel is open closes it 
* cleanup code for adding a conditional bp
* simplify cbPanel
* selecting "add bp" with an open panel closes the panel
* tests
...

### CODE REVIEW

**Comment by jasonLaster** [comment]
@jlongster should be fixed.

File: public/js/test/mochitest/head.js [hunk 1]
```diff
@@ -559,6 +563,7 @@ function pressKey(dbg, keyName) {
 
 function type(dbg, string) {
   string.split("").forEach(char => {
+    console.log("typing character", char)
```
Comment (jlongster) [inline public/js/test/mochitest/head.js]: Leftover console.log

File: public/js/components/Editor.js [hunk 2]
```diff
@@ -255,7 +260,7 @@ const Editor = React.createClass({
   },
 
   componentDidMount() {
-    this.cbPanels = {};
+    this.cbPanel;
```
Comment (jlongster) [inline public/js/components/Editor.js]: This line doesn't do anything

Comment (jasonLaster) [inline public/js/components/Editor.js]: haha oops, that should be `this.cbPanel = null`.

**Pull request merged**